### ISSUE TITLE
Indirect Code Note Crash

### ISSUE THREAD

**Issue by SporyTike** [RetroAchievements/RAIntegration#956]
While developing with pointers I noticed some crashes related to indirect code notes. I currently did not find a way to recreate it everytime, but it seems to be related to the process where the toolkit creates the indirect code note.

Since I didn't find a recreation method that works all the time I will explain what I did to encounter the issue in the first place with the hope that either some other developer encounters this issue too and can give an actual recreation method or maybe the info is already enough to figure out what could be wrong.

So I have a pointer note which is simply:
```
Pointer - Player Information [32bit]
+0x0 | Test 1
+0x1 | Test 2
+0x2 | Test 3
+0x3 | Test 4
+0x4 | Test 5
+0x5 | Test 6
```
Sometimes when adding a new offset to the note the emulator just crashes as soon you typed the `+0x`. My assumption would be that it tries to transfer `+0x` into an address and to place a note there which obviously isn't possible, but I can't say for sure since it only happens sometimes.

Also not sure if it's just me being unlucky while testing or if it's actually related to the issue but it only crashed under above's criteria when it was shortly after loading the game, publishing or reverting the note.

Sorry that I can't give much for info to it, I tested for like an hour and wasn't able to find a consistent recreation method. If further help is needed, I am down to help.

**Comment by Jamiras** [comment]
I've spent about 20 minutes trying to reproduce this myself, without luck. The partial offset appears correctly in the Code Note Window and the tooltip in the achievement editor updates correctly when modifying the new line, both when the condition offset is the line being modified and when it's not.

When you reopen the game are the modifications present? (did the `0x` get written to the xxx-user file before it crashed?)

**Comment by SporyTike** [comment]
No, seems like it does not get written into the xxx-user file. When reopening it it's the same as it was before.

I tried a bit and got a recording of the crash in case it could be helpful. As you can see in the video I also managed to type another character behind the `0x` so whatever crashes the emulator, it seems to give a little bit of time before it crashes.

https://user-images.githubusercontent.com/45672834/211211008-b520200c-68df-4479-a48a-e4167da39651.mp4
